CMakeLists.txt

```diff
@@ -0,0 +1,143 @@
+# Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.13.4)
+
+# TODO: Handle BUILD_VERSION for devops
+# set(CHPL_BUILD_VERSION 0 CACHE ) - gets into generated.h along with version info,
+
+# TODO: Write this to generated configured_prefix file
+# CMAKE_INSTALL_PREFIX
+
+project(Chapel VERSION 1.29.0)
+
+# This built-in module allows us to print variables easily, using
+# cmake_print_variables(MY_VARIABLE)
+include(CMakePrintHelpers)
+
+# option(CHPL_LLVM_COMP_ARGS "Override compile args for LLVM")
+# option(CHPL_LLVM_LINK_ARGS "Override linker args for LLVM")
+set(CHPL_HOME CACHE STRING "Override home directory for Chapel")
+
+if (NOT DEFINED ${CHPL_HOME})
+  message(STATUS "CHPL_HOME not set, using $ENV{CHPL_HOME}")
+  set(CHPL_HOME $ENV{CHPL_HOME} CACHE STRING "Override home directory for Chapel" FORCE)
+endif()
+
+# Options that correspond to printchplenv --internal --compiler
+# Could be set from environment or command-line
+# Print the output of PRINTCHPLENV to user
+set(CHPL_HOST_BUNDLED_COMPILE_ARGS CACHE STRING "Description")
```

Comment:
The only ones that should not be internal are these

```
$ ./util/printchplenv  --compiler
CHPL_HOST_PLATFORM: linux64
CHPL_HOST_COMPILER: gnu
  CHPL_HOST_CC: gcc
  CHPL_HOST_CXX: g++
CHPL_HOST_ARCH: x86_64
CHPL_HOST_MEM: cstdlib
CHPL_LLVM: system *
  CHPL_LLVM_SUPPORT: system
  CHPL_LLVM_CONFIG: llvm-config-14
  CHPL_LLVM_VERSION: 14
CHPL_SANITIZE: none
```

(But actually, CHPL_LLVM_VERSION is not something a user can set, so that should be internal. I think a user cat set the rest of these.).

CHPL_HOST_BUNDLED_COMPILE_ARGS etc is internal.

CHPL_GMP is only used for the runtime, so not actually relevant in our cmake files yet. (But harmless if you keep it internal).